src/buffer/out/textBuffer.cpp

```diff
@@ -1033,238 +1036,172 @@ const TextBuffer::TextAndColor TextBuffer::GetTextForClipboard(const bool lineSe
-
-        std::string const szHtmlFragStart = "<!--StartFragment -->";
-        std::string const szHtmlFragEnd = "<!--EndFragment -->";
-        std::string const szHtmlFooter = "</BODY></HTML>";
-        size_t const cbHtmlFooter = szHtmlFooter.size();
-
-        std::string const szDivOuterBackgroundPattern = R"X(<DIV STYLE="background-color:#%02x%02x%02x;white-space:pre;">)X";
-
-        size_t const cbDivOuter = 55;
-        std::string szDivOuter;
-        szDivOuter.reserve(cbDivOuter);
-
-        std::string const szSpanFontSizePattern = R"X(<SPAN STYLE="font-size: %dpt">)X";
-
-        size_t const cbSpanFontSize = 28 + (iFontHeightPoints / 10) + 1;
-
-        std::string szSpanFontSize;
-        szSpanFontSize.resize(cbSpanFontSize + 1); // reserve space for null after string for sprintf
-        sprintf_s(szSpanFontSize.data(), cbSpanFontSize + 1, szSpanFontSizePattern.data(), iFontHeightPoints);
-        szSpanFontSize.resize(cbSpanFontSize); //chop off null at end
-
-        std::string const szSpanStartPattern = R"X(<SPAN STYLE="color:#%02x%02x%02x;background-color:#%02x%02x%02x">)X";
-
-        size_t const cbSpanStart = 53; // when format is expanded, there will be 53 bytes per color pattern.
-        std::string szSpanStart;
-        szSpanStart.resize(cbSpanStart + 1); // +1 for null terminator
+        std::ostringstream htmlBuilder;
 
-        std::string const szSpanStartFontPattern = R"X(<SPAN STYLE="font-family: '%s', monospace">)X";
-        size_t const cbSpanStartFontPattern = 41;
-
-        std::string const szSpanStartFontConstant = R"X(<SPAN STYLE="font-family: monospace">)X";
-        size_t const cbSpanStartFontConstant = 37;
+        // First we have to add some standard
+        // HTML boiler plate required for CF_HTML
+        // as part of the HTML Clipboard format
+        constexpr std::string_view HtmlHeader =
+            "<!DOCTYPE><HTML><HEAD><TITLE>Windows Terminal</TITLE></HEAD><BODY>";
+        htmlBuilder << HtmlHeader;
 
-        std::string szSpanStartFont;
-        size_t cbSpanStartFont;
-        bool fDeleteSpanStartFont = false;
+        htmlBuilder << "<!--StartFragment -->";
 
-        std::wstring const wszFontFaceName = fontFaceName;
-        size_t const cchFontFaceName = wszFontFaceName.size();
-        if (cchFontFaceName > 0)
+        // apply global style in div element
         {
-            // measure and create buffer to convert face name to UTF8
-            int const cbNeeded = WideCharToMultiByte(CP_UTF8, 0, wszFontFaceName.data(), static_cast<int>(cchFontFaceName), nullptr, 0, nullptr, nullptr);
-            std::string szBuffer;
-            szBuffer.resize(cbNeeded);
+            htmlBuilder << "<DIV STYLE=\"";
+            htmlBuilder << "display:inline-block";
+            htmlBuilder << "width:400px;";
+            htmlBuilder << "white-space:pre;";
+
+            // fixme: this is only walkaround for filling background after last char of row.

```

Comment:
Could you please create a bug report on GitHub and reference it in the comment? That way we can track it and know to go here to fix it.